### PR TITLE
chore(initial checkin): add artifacts required to start off this project

### PR DESCRIPTION
**Pull request by AmitKumarDas**
This commit adds the following artifacts required to proceed with coding:

- Dockerfile to create image
- Makefile to provide various build related targets
- Go vendoring via go modules
- Some .go files to help during build

Signed-off-by: AmitKumarDas <amit.das@mayadata.io>

### CODE REVIEW

**Review by ksatchit — APPROVED** [review]
lgtm

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# re-download as much and so that source changes don't invalidate our 
+# downloaded layer
+RUN make vendor
+
+# copy go source code
+COPY cmd/ cmd/
+COPY pkg/ pkg/
+
+# build the binary
+RUN make lbins
```
Comment (utkarshmani1997) [inline Dockerfile:24]: Good that we are building this inside container, we no longer have to depend on the installation of updated version of go locally, only image change is required.

**Review by utkarshmani1997 — APPROVED** [review]

File: pkg/utils/build/build.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (c) 2019 The MayaData Authors.
+Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (satyamz) [inline pkg/utils/build/build.go:3]: Is this intentional? :thinking:

Comment (vishnuitta) [reply pkg/utils/build/build.go:3]: does this needs the fix @AmitKumarDas ? in other files also

Comment (AmitKumarDas) [reply pkg/utils/build/build.go:3]: This is required. Parts of the file were from `https://github.com/uber/makisu`

cc @satyamz @vishnuitta

Comment (satyamz) [reply pkg/utils/build/build.go:3]: > This is required. Parts of the file were from https://github.com/uber/makisu

Nice, I am in favor of keeping it. :hugs:

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Build the operator binary
+FROM golang:1.12.5 as builder
```
Comment (AmitKumarDas) [inline Dockerfile:2]: Can we use latest go stable version

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+.PHONY: test unit-test
+test: unit-test
+
+unit-test: $(ALL_SRC) vendor ext-tools
+	$(EXT_TOOLS_DIR)/gocov test $(ALL_PKGS) --tags "unit" | $(EXT_TOOLS_DIR)/gocov report
+
+gofmt:
+	@go fmt $(ALL_PKG_PATHS)
+
+lint: ext-tools gofmt
```
Comment (AmitKumarDas) [inline Makefile:99]: Please verify if these work with go modules

File: go.mod [hunk 5]
```diff
@@ -0,0 +1,76 @@
+module github.com/mayadata-io/openebs-operator
+
+go 1.12
```
Comment (AmitKumarDas) [inline go.mod:3]: Let us use latest go version

File: pkg/apis/mayadata.io/openebs/v1alpha1/operator.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (AmitKumarDas) [inline pkg/apis/mayadata.io/openebs/v1alpha1/operator.go:24]: What is the controller library that this project will be using? Is it operator sdk or kubebuilder? If its metac then code gen is not required.

**Pull request merged**